### PR TITLE
Add histogram to dataset settings

### LINKED ISSUE #3583: (Auto) min-max to determine brightness / contrast
The brightness / contrast sliders are equal to a min and max value. We could think about exposing these values directly via the UI (one user requested this).

Even better would be an automatic determination of appropriate min/max values. This could work by inspecting the data within a bounding box and building a histogram to derive these values.

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds the long-awaited histogram feature for to the dataset settings. It uses the new histogram backend route to get the sampled histogram data of the current dataset.

### URL of deployed dev instance (used for testing):
- [https://addhistogramtodatasetsettings.webknossos.xyz](https://addhistogramtodatasetsettings.webknossos.xyz)

### Steps to test:
- Open some tracing
- There open the dataset settings
- Each layer should have its own histogram. With this, you can adjust the max and min value.

### Issues:
- fixes #3583

### ToDo:
- [x] Discuss what to do the [legacy options](https://github.com/scalableminds/webknossos/blob/586a7f4cd83ecc9b1456fc1670cbf7ce56d259b7/frontend/javascripts/libs/datasource.schema.json#L143) of brightness and contrast. 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
@MichaelBuessemeyer can you add the remaining to dos in form of checkboxes to this PR? Or is everything done and ready for review? :)

**Comment by MichaelBuessemeyer** [comment]
@philippotto As we can't convert brightness and contrast into a good approximation of the intensityRange (min and max value) and we no longer support brightness and contrast: Do you thing just replacing brightness and contrast in the dataset layer settings is fine? I think @daniel-wer also suggested this. And using default values should be fine as our customer mentioned this in the discuss :slightly_smiling_face: 

Maybe last TODOs:
- [ ] Automatically replace the brightness and contrast parameter in the default dataset layer settings of a dataset with an `intensityRange` of [0, 255]
- [ ] Add intensityRange to the required values for valid dataset layer settings

**Comment by philippotto** [comment]
> As we can't convert brightness and contrast into a good approximation of the intensityRange (min and max value) and we no longer support brightness and contrast: Do you thing just replacing brightness and contrast in the dataset layer settings is fine? I think @daniel-wer also suggested this. And using default values should be fine as our customer mentioned this in the discuss slightly_smiling_face

Ok, sounds good!

> - [ ] Automatically replace the brightness and contrast parameter in the default dataset layer settings of a dataset with an intensityRange of [0, 255]
> - [ ]  Add intensityRange to the required values for valid dataset layer settings

Regarding the second todo: the value should be valid and accepted, but it shouldn't be required. Otherwise, all existing dataset layer settings will become immediately invalid when we deploy this. Instead, the old configurations should still be valid (so, the old brightness/contrast values are still valid and ignored and the new intensityRange is "optional").

File: frontend/javascripts/libs/datasource.types.js [hunk 1]
```diff
@@ -4,8 +4,9 @@
 // The `flow2schema` node module has been used for conversion.
 // Please note that some manual changes to the schema are required.
 
-// Add minItems=3 and maxItems=3
-type Vector3 = Array<number>;
+type Vector3 = [number, number, number];
+
+type Vector2 = [number, number];
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/libs/datasource.types.js]: I think defining the Vectors as tuples is more accurate as it defines the length of the array

File: frontend/stylesheets/trace_view/_user_settings_view.less [hunk 2]
```diff
@@ -5,3 +5,7 @@
 .centered-children {
   text-align: center;
 }
+
+.tracing-settings-menu .ant-tabs-content {
+  padding-right: 8px;
```
Comment (MichaelBuessemeyer) [inline frontend/stylesheets/trace_view/_user_settings_view.less]: I added some padding to the settings on the left to not get elements covered by the scrolling bar.

**Comment by MichaelBuessemeyer** [comment]
@philippotto This PR is ready for your review :tada:

**Comment by MichaelBuessemeyer** [comment]
> - [x] For float layers, we said that we won't support the histogram in the first iteration, right? Then, I'd hide the histogram for those layers. Right now, the user can use the slider, but it won't change anything. Also, the histogram doesn't look right due to the range of 0-255.
> - [x] When disabling a layer, the min-max-slider should be disabled (and I'd argue that the histogram could also be hidden)

@philippotto Thanks for paying attention to those points.
Regarding the second point, I also agree that if you have a layer disabled, the histogram of that layer should be hidden. This makes the settings more compact.

Please note: As I do not have a dataset with float type I cannot try out the first point.

And you can give this PR another of your awesome reviews :)

**Comment by philippotto** [comment]
@MichaelBuessemeyer This can be merged, right?

**Comment by MichaelBuessemeyer** [comment]
@philippotto Indeed, It can.
I can't work on possible fixes today, but this should be alright :slightly_smiling_face: 
And thanks for updating this branch

**Pull request merged**